platform/source/mbed_retarget.cpp

```diff
@@ -1933,3 +1933,268 @@ extern "C" MBED_WEAK long int _scanf_mbtowc(
+            break;
+        case EREMOTE:
+            break;
+        case ENOLINK:
+            break;
+        case EADV:
+            break;
+        case ESRMNT:
+            break;
+        case ECOMM:
+            break;
+        case EPROTO:
+            break;
+        case EMULTIHOP:
+            break;
+        case EDOTDOT:
+            break;
+        case EBADMSG:
+            break;
+        case EOVERFLOW:
+            break;
+        case ENOTUNIQ:
+            break;
+        case EBADFD:
+            break;
+        case EREMCHG:
+            break;
+        case ELIBACC:
+            break;
+        case ELIBBAD:
+            break;
+        case ELIBSCN:
+            break;
+        case ELIBMAX:
+            break;
+        case ELIBEXEC:
+            break;
+        case EILSEQ:
+            break;
+        case ERESTART:
+            break;
+        case ESTRPIPE:
+            break;
+        case EUSERS:
+            break;
+        case ENOTSOCK:
+            break;
+        case EDESTADDRREQ:
+            break;
+        case EMSGSIZE:
+            break;
+        case EPROTOTYPE:
+            break;
+        case ENOPROTOOPT:
+            break;
+        case EPROTONOSUPPORT:
+            break;
+        case ESOCKTNOSUPPORT:
+            break;
+        case EOPNOTSUPP:
+            break;
+        case EPFNOSUPPORT:
+            break;
+        case EAFNOSUPPORT:
+            break;
+        case EADDRINUSE:
+            break;
+        case EADDRNOTAVAIL:
+            break;
+        case ENETDOWN:
+            break;
+        case ENETUNREACH:
+            break;
+        case ENETRESET:
+            break;
+        case ECONNABORTED:
+            break;
+        case ECONNRESET:
+            break;
+        case ENOBUFS:
+            break;
+        case EISCONN:
+            break;
+        case ENOTCONN:
+            break;
+        case ESHUTDOWN:
+            break;
+        case ETOOMANYREFS:
+            break;
+        case ETIMEDOUT:
+            break;
+        case ECONNREFUSED:
+            break;
+        case EHOSTDOWN:
+            break;
+        case EHOSTUNREACH:
+            break;
+        case EALREADY:
+            break;
+        case EINPROGRESS:
+            break;
+        case ESTALE:
+            break;
+        case EUCLEAN:
+            break;
+        case ENOTNAM:
+            break;
+        case ENAVAIL:
+            break;
+        case EISNAM:
+            break;
+        case EREMOTEIO:
+            break;
+        case EDQUOT:
+            break;
+        case ENOMEDIUM:
+            break;
+        case EMEDIUMTYPE:
+            break;
+        case ECANCELED:
+            break;
+        case ENOKEY:
+            break;
+        case EKEYEXPIRED:
+            break;
+        case EKEYREVOKED:
+            break;
+        case EKEYREJECTED:
+            break;
+        case EOWNERDEAD:
+            break;
+        case ENOTRECOVERABLE:
+            break;
```

Comment:
I think we should check for equality too as pointed out by Kevin. 

can you add static asserts for them ?

Reply:
Eh? What did I suggest? The switch is testing for equality. (A comment explaining that would be a good idea...)

Not sure the `break`s are needed - wouldn't expect any compiler to complain about "fallthrough" for directly adjacent `case`s.

Reply:
I should have precise that I was speaking about `EAGAIN` and `EWOULDBLOCK`. It is not mandatory to have them equal thought. 
In such case we need a second switch case where `EAGAIN` is replaced with `EWOULDBLOCK`.

Reply:
I believe you can simply do:

    #if EAGAIN != EWOULDBLOCK
    case EAGAIN:
    #endif
    case EWOULDBLOCK: 

[POSIX says](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/errno.h.html) the defines have to be usable in `#if` directives.